src/background/background.ts

```diff
@@ -12,11 +12,40 @@ import { vaultMessageHandler } from '@background/vault';
 import { IS_TEST_ENV } from '@common/constants';
 import { CONTENT_SCRIPT_PORT } from '@content-scripts/content-script';
 import { VaultActions } from '@background/vault-types';
-import { ExternalMethods, MessageFromContentScript } from '@content-scripts/message-types';
+import {
+  ExternalMethods,
+  InternalMethods,
+  MessageFromContentScript,
+  MESSAGE_SOURCE,
+} from '@content-scripts/message-types';
+
+function setAppParamOfSourceInstallingExtension() {
+  // The app url is saved for use after onboarding
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const appUrl = urlParams.get('app');
```

Comment:
`'app`` i think should be less specific, eg `redirectUrl`. also i think it should be a const/enum

```ts
const REDIRECT_PARAM = 'redirectUrl`
```